core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -318,6 +323,16 @@ private Object processWithExceptionHandling(Expression expression, Object contex
             }
         }
 
+        private boolean canUseCache(Object context)
+        {
+            // We are using WeakHashMap as cache, that's why we can't depend on an intermediate object
+            // that consists of expression as well as context as a key. This is because intermediate object could
+            // get GCed even when expression still exists which will cause cache miss. So, to solve this problem we
```

Comment:
the comment about  GC and weak keys belongs to `optimizationCache` field declaration (or `optimizationCache` usage). it doesn't belong to `canUseCache` method